sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -226,6 +256,7 @@ public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
                     status.getTaskId().getValue(), optionalTaskInfo.get().getTaskId().getValue(),
                     status, optionalTaskInfo));
         }
+        /*  All checks till here are redundant if we get TaskName via looking at matching TaskIDs */
```

Comment:
I don't understand this comment.  If we have redundant checks they should be removed.